### PR TITLE
refactor: forester + tests

### PR DESCRIPTION
**Pull request by sergeytimoshin**
• use only one rpc instance in forester
• `create_and_send_transaction_with_event`  return signature along with event 
• add test helpers to test_env

### CODE REVIEW

File: test-utils/src/rpc/solana_rpc.rs
Comment (SwenSchaeferjohann) [inline test-utils/src/rpc/solana_rpc.rs:190]: think we'll need to enable these again for merging

File: forester/tests/test_indexer.rs
Comment (SwenSchaeferjohann) [inline forester/tests/test_indexer.rs:74]: do we still need this test after #810  gets merged?

File: forester/src/nullifier/nullify.rs
Comment (ananas-block) [inline forester/src/nullifier/nullify.rs:45]: This will log a lot of data.

File: forester/tests/nullifier_test.rs
Comment (sergeytimoshin) [reply forester/tests/nullifier_test.rs:37]: No, it's already there, right on the next line, we reuse this rpc connection through the forester internals:

 https://github.com/Lightprotocol/light-protocol/pull/811/files#diff-b1dc72fd951f332ca2b32620374dbe4f2c407c91a57c91943e43cdcd627b2a17R38

File: package.json
Comment (sergeytimoshin) [reply package.json]: 100% not needed, weird leftover

File: forester/src/indexer/photon_indexer.rs
Comment (SwenSchaeferjohann) [inline forester/src/indexer/photon_indexer.rs:100]: leftover?

File: test-programs/e2e-test/tests/test.rs
Comment (SwenSchaeferjohann) [inline test-programs/e2e-test/tests/test.rs:29]: Iirc, @ananas-block discovered that we can use a random keypair here for the payer as a hotfix. The underlying issue is that the same hardcoded keypair gets used elsewhere in a completely unrelated test at the same time, and therefore, the airdrop here updates its balance during that other test run, causing that other test to fail.

File: pnpm-lock.yaml
Comment (SwenSchaeferjohann) [inline pnpm-lock.yaml:1]: any idea why this is happening?

File: test-programs/e2e-test/tests/test.rs
Comment (SwenSchaeferjohann) [inline test-programs/e2e-test/tests/test.rs:29]: same here.

**Review by SwenSchaeferjohann — COMMENTED** [review]
just these notes here - otherwise lgtm!

File: forester/tests/test_indexer.rs
Comment (ananas-block) [reply forester/tests/test_indexer.rs:37]: agreed we should use confirm to wait for confirmation

Comment (sergeytimoshin) [reply forester/tests/test_indexer.rs]: Thanks! Fixed.

File: pnpm-lock.yaml
Comment (sergeytimoshin) [reply pnpm-lock.yaml:1]: desperate attempt to fix weird ci cache issue by calling pnpm update

**Review by SwenSchaeferjohann — COMMENTED** [review]
thx lgtm, any rem. thoughts @ananas-block ?

File: test-utils/src/spl.rs
Comment (ananas-block) [inline test-utils/src/spl.rs:62]:
```suggestion
    let (event, _) = rpc
```
this works as well and produces less noise.

File: test-utils/src/rpc/test_rpc.rs
Comment (ananas-block) [inline test-utils/src/rpc/test_rpc.rs:28]: A result struct would be a lot more intuitive than a tuple in my opinion.
Let's do that in a separate pr though.

File: test-utils/src/rpc/solana_rpc.rs
Comment (ananas-block) [inline test-utils/src/rpc/solana_rpc.rs:277]: Also no need to do that in this pr but we need to get rid of these loops without a timeout.
If the transaction fails this will result in an infinite loop.

File: scripts/install.sh
Comment (ananas-block) [inline scripts/install.sh:113]: could this cause the pnpm lock changes?

File: forester/tests/test_indexer.rs
Comment (ananas-block) [inline forester/tests/test_indexer.rs:65]: why not use the result here?
We are awaiting the transaction so I don't see why we shouldn't fail if the transaction fails.

File: forester/tests/nullifier_test.rs
Comment (ananas-block) [inline forester/tests/nullifier_test.rs:99]: same here why not throw an error if the airdrop fails?

File: forester/tests/nullifier_test.rs
Comment (ananas-block) [inline forester/tests/nullifier_test.rs:120]: and here

File: forester/src/nullifier/nullify.rs
Comment (ananas-block) [inline forester/src/nullifier/nullify.rs:100]: on testnet this retry logic will not work. It will hit the rpc ratelimit immediately.
You will need a delay for retries.

Comment (ananas-block) [reply forester/src/nullifier/nullify.rs:100]: It probably makes sense to create a function send with retries.

**Review by ananas-block — COMMENTED** [review]
Got a couple comments.
The only thing that I would want to discuss and potentially fix in this pr are the unused results.
(``let _ = function_that_returns_result().await;``)

File: scripts/install.sh
Comment (sergeytimoshin) [reply scripts/install.sh:113]: No. pnpm-lock was updated when I called pnpm update.

File: test-utils/src/rpc/solana_rpc.rs
Comment (sergeytimoshin) [reply test-utils/src/rpc/solana_rpc.rs:277]: Hm, as far as I understand, If the transaction fails, it will break the loop due to the error propagation through `?`. 
The only theoretical possibility of an infinite loop: transaction never confirmed.

Also, https://github.com/anza-xyz/agave/blob/master/rpc-client/src/rpc_client.rs#L979-L985

File: forester/src/nullifier/nullify.rs
Comment (sergeytimoshin) [reply forester/src/nullifier/nullify.rs:100]: It has the delay here: https://github.com/Lightprotocol/light-protocol/pull/811/files/44d4ead2d6dc267dfbb4511305667fe0cd125aaa#diff-fed5e22bba6202e7a68545ef7f19570dbd3c084c5aa2f0d46703fd176a94371cR138

File: forester/tests/nullifier_test.rs
Comment (sergeytimoshin) [reply forester/tests/nullifier_test.rs]: Thanks, fixed!

File: forester/tests/nullifier_test.rs
Comment (sergeytimoshin) [reply forester/tests/nullifier_test.rs]: Fixed.

File: forester/tests/test_indexer.rs
Comment (sergeytimoshin) [reply forester/tests/test_indexer.rs]: Fixed.

File: scripts/install.sh
Comment (ananas-block) [reply scripts/install.sh:113]: ah I see

File: test-utils/src/rpc/solana_rpc.rs
Comment (ananas-block) [reply test-utils/src/rpc/solana_rpc.rs:277]: I have been stuck in a loop like this when testing a couple days ago.
In the case where I was stuck, the rpc returned transaction signature even though tx has failed and I was able to look at the failed tx in the explorer.

Comment (ananas-block) [reply test-utils/src/rpc/solana_rpc.rs:277]: Try this logic with transaction that does a transfer from an account with insufficient funds.

File: forester/tests/empty_address_tree_test.rs
Comment (ananas-block) [inline forester/tests/empty_address_tree_test.rs:80]: one more unused result

File: test-utils/src/rpc/solana_rpc.rs
Comment (sergeytimoshin) [reply test-utils/src/rpc/solana_rpc.rs:277]: Gotcha, thanks. I wrote a note for myself, let's fix this on another PR.

File: test-utils/src/spl.rs
Comment (ananas-block) [reply test-utils/src/spl.rs:62]: agreed just saying

**Review by ananas-block — APPROVED** [review]

**Pull request merged**
